arm/Microsoft.Network/applicationGateways/deploy.bicep

```diff
@@ -151,244 +240,108 @@ var diagnosticsMetrics = [for metric in metricsToEnable: {
+resource applicationGateway 'Microsoft.Network/applicationGateways@2021-05-01' = {
   name: name
   location: location
-  identity: identity
   tags: tags
-  properties: {
-    sku: {
-      name: sku
-      tier: endsWith(sku, 'v2') ? sku : substring(sku, 0, indexOf(sku, '_'))
-      capacity: capacity
-    }
-    gatewayIPConfigurations: [
-      {
-        name: gatewayIpConfigurationName
-        properties: {
-          subnet: {
-            id: subnetResourceId
-          }
-        }
-      }
-    ]
-    frontendIPConfigurations: [
-      {
-        name: frontendPrivateIPConfigurationName
-        type: 'Microsoft.Network/applicationGateways/frontendIPConfigurations'
-        properties: empty(frontendPrivateIpAddress) ? frontendPrivateIPDynamicConfiguration : frontendPrivateIPStaticConfiguration
-      }
-      {
-        name: frontendPublicIPConfigurationName
-        properties: {
-          publicIPAddress: {
-            id: frontendPublicIpResourceId
-          }
-        }
-      }
-    ]
-    sslCertificates: empty(sslCertificateKeyVaultSecretId) ? null : sslCertificates
+  identity: identity
+  properties: union({
+    authenticationCertificates: authenticationCertificates
+    autoscaleConfiguration: autoscaleMaxCapacity > 0 && autoscaleMinCapacity > 0 ? {
+      maxCapacity: autoscaleMaxCapacity
+      minCapacity: autoscaleMinCapacity
+    } : null
     backendAddressPools: backendAddressPools
-    probes: probes_var
-    backendHttpSettingsCollection: backendHttpConfigurations_var
+    backendHttpSettingsCollection: backendHttpSettingsCollection
+    customErrorConfigurations: customErrorConfigurations
+    enableHttp2: enableHttp2
+    firewallPolicy: !empty(firewallPolicyId) ? {
+      id: firewallPolicyId
+    } : null
+    forceFirewallPolicyAssociation: !empty(firewallPolicyId)
+    frontendIPConfigurations: frontendIPConfigurations
     frontendPorts: frontendPorts
+    gatewayIPConfigurations: gatewayIPConfigurations
+    globalConfiguration: {
+      enableRequestBuffering: enableRequestBuffering
+      enableResponseBuffering: enableResponseBuffering
+    }
     httpListeners: httpListeners
+    loadDistributionPolicies: loadDistributionPolicies
+    privateLinkConfigurations: privateLinkConfigurations
+    probes: probes
     redirectConfigurations: redirectConfigurations
     requestRoutingRules: requestRoutingRules
-    enableHttp2: http2Enabled
-    webApplicationFirewallConfiguration: startsWith(sku, 'WAF') ? wafConfiguration : null
-  }
-  dependsOn: []
+    rewriteRuleSets: rewriteRuleSets
+    sku: {
+      name: sku
+      tier: endsWith(sku, 'v2') ? sku : substring(sku, 0, indexOf(sku, '_'))
+      capacity: autoscaleMaxCapacity > 0 && autoscaleMinCapacity > 0 ? null : capacity
+    }
+    sslCertificates: sslCertificates
+    sslPolicy: {
+      cipherSuites: sslPolicyCipherSuites
+      // disabledSslProtocols: sslPolicyDisabledSslProtocols
```

Comment:
Is this to be removed?

Reply:
Yes. Not supported anymore. Will remove it.

Reply:
done